File: app/src/main/java/com/executivestrokes/githubreposissues/IssuesActivity.java
package com.executivestrokes.githubreposissues;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

import com.executivestrokes.Adapters.IssueAdapter;
import com.executivestrokes.RestApi.IssueInterface;
import com.executivestrokes.RestApi.RetrofitCall;
import com.executivestrokes.model.IssueModel;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class IssuesActivity extends AppCompatActivity {
    String owner;
    String repo;
    TextView txt;
    RecyclerView mRecyclerView;
    List<IssueModel> issueData = new ArrayList<>();
    RecyclerView.Adapter issuAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_issues);
        txt=findViewById(R.id.textView);
        Bundle extras = getIntent().getExtras();
        owner = extras.getString("USERS");
        repo = extras.getString("repoTitle");
        txt.setText("User : "+owner+"\nRepository : "+repo);

        //Adapter and Recycler
        mRecyclerView=  findViewById(R.id.issue_recycler_view);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        issuAdapter = new IssueAdapter(issueData, R.layout.list_item_issues,
                getApplicationContext());

        mRecyclerView.setAdapter(issuAdapter);
        //Instance of interface
        IssueInterface issueInterface= RetrofitCall.getUserInfo().create(IssueInterface.class);
       if (owner!=null && repo!=null)
        {
            Call<List<IssueModel>> call = issueInterface.getIssues(owner,repo);
            call.enqueue(new Callback<List<IssueModel>>() {
                @Override
                public void onResponse(Call<List<IssueModel>> call, Response<List<IssueModel>> response) {
                    if(response.isSuccessful())
                    {
                        issueData.clear();
                        issueData.addAll(response.body());
                        issuAdapter.notifyDataSetChanged();
                    }
                    else
                    {
                        Toast.makeText(IssuesActivity.this,"Error ! No Issues found for the given Repository",Toast.LENGTH_SHORT).show();
                    }
                }

                @Override
                public void onFailure(Call<List<IssueModel>> call, Throwable t) {
                    System.out.println("Failed!" + t.toString());
                }
            });
        }
        else
        {
            Toast.makeText(IssuesActivity.this,"Error ! Getting User and Repository name...",Toast.LENGTH_SHORT).show();
        }



    }
}

File: github_app/app/src/main/java/com/example/github/issues/IssuesActivity.kt
package com.example.github.issues

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.github.R
import com.example.github.adapter.IssuesAdapter
import com.example.github.databinding.ActivityIssuesBinding
import com.example.github.factory.IssuesViewModelFactory
import com.example.github.models.Issue
import com.example.github.repositories.IssuesRepositories
import com.example.github.viewModels.IssuesViewModel

class IssuesActivity : AppCompatActivity() {
    private lateinit var binding: ActivityIssuesBinding
    private lateinit var viewModel: IssuesViewModel

    private var list = ArrayList<Issue>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityIssuesBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val repository = IssuesRepositories()
        val viewModelFactory = IssuesViewModelFactory(repository)
        viewModel = ViewModelProvider(this, viewModelFactory).get(IssuesViewModel::class.java)

    }

    @SuppressLint("NotifyDataSetChanged")
    override fun onStart() {
        super.onStart()

        val linearLayoutManager = LinearLayoutManager(
            this@IssuesActivity,
            LinearLayoutManager.VERTICAL,
            false
        )

        var dir = ""
        val intent = intent
        if (intent != null) {
            if (intent.hasExtra("dir")) {
                dir = intent.getStringExtra("dir").toString()
            }

            if (intent.hasExtra("name") && intent.hasExtra("repo")) {

                viewModel.getIssues(
                    intent.getStringExtra("name")!!,
                    intent.getStringExtra("repo")!!
                )

                binding.repoName.text = intent.getStringExtra("repo")
            }
        }

        val adapter = IssuesAdapter(list, this@IssuesActivity, dir)
        binding.recyclerIssues.layoutManager = linearLayoutManager
        binding.recyclerIssues.adapter = adapter

        viewModel.issue.observe(this, { response ->
            if (response.isSuccessful) {
                if (response.body() != null && response.body()!!.isNotEmpty()) {
                    list.clear()
                    list.addAll(response.body()!!)
                    adapter.notifyDataSetChanged()
                }
            } else {
                Toast.makeText(this@IssuesActivity, response.code().toString(), Toast.LENGTH_SHORT).show()
            }
        })

        binding.back.setOnClickListener {
            this@IssuesActivity.finish()
        }
    }
}